### ISSUE TITLE
[BUG] Did not read the aria-label attribute for button element

### ISSUE THREAD

**Issue by ttliang** [IBMa/equal-access#475]
## What tool(s) are you using?

- [ ] `accessibility-checker-engine`
- [x] `accessibility-checker`
- [ ] `karma-accessibility-checker`
- [ ] `rule-server`
- [ ] `boilerplates`
- [ ] `other`

## Detailed description

> Platform: MacOS 11.4, Firefox: 90.0.2(64bit)

> Describe in detail the issue you're having.
button element has 'aria-label' attribute while AVT ff extension did report a violation "Accessible name does not match or contain the visible label text".

**Screencut:**
![image](https://user-images.githubusercontent.com/12162531/129175774-87963ecb-324b-4f6e-9bc6-4dc2308a6ed0.png)
**Code:**
`<div class="bx--col-sm-4 swu-export-menu" style="flex: 33%;"><ibm-overflow-menu><button aria-haspopup="true" class="bx--overflow-menu bx--overflow-menu--open" type="button" aria-label="Overflow" aria-expanded="true" role="button"><!----><!----><span role="button"><button ibmbutton="ghost" class="bx--btn bx--btn--ghost"> Export <svg class="bx--btn__icon"><use xlink:href="assets/carbon-icons.svg#icon--download"></use></svg></button></span><!----></button><!----><!----></ibm-overflow-menu></div>`

**Comment by tombrunet** [comment]
This is invalid HTML. Buttons cannot be nested. As a result, the parsers break that hierarchy up in all sorts of strange ways. What is the 'visible text' is therefore undefined, and in this case, it's probably picking up all of the contents. This is not an engine error, but rather an incorrectly coded page.
